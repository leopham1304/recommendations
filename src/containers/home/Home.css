.autocomplete-wrapper {
  display: flex;
  justify-content: center;
  margin-top: 20vh;
}

.home-container {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: url(../../assets/images/background.svg);
  height: 100%;
  justify-content: center;
}
