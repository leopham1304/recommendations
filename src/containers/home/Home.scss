.home-container {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: url(../../assets/images/background.jpg);
  height: 100%;
  justify-content: center;
}

.dark-mode {
  box-shadow: inset 0 0 0 1000px rgba(0, 0, 0, 0.85);
}

.autocomplete-wrapper {
  display: flex;
  justify-content: center;
  margin-top: 20vh;
}
