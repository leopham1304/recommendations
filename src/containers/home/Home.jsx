import React, { memo } from "react";
import { Autocomplete, NavigationBar } from "../../components";
import { useHome } from "./Home.utils";
import "./Home.scss";

const Home = () => {
  const { darkMode, setDarkMode } = useHome();

  return (
    <div className={darkMode ? "home-container dark-mode" : "home-container"}>
      <NavigationBar setDarkMode={setDarkMode} darkMode={darkMode} />
      <div className="autocomplete-wrapper">
        <Autocomplete darkMode={darkMode} />
      </div>
    </div>
  );
};

export default memo(Home);
