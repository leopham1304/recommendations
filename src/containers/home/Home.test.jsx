import React from "react";
import Home from "./Home";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Home", () => {
  const wrapper = shallow(<Home />);

  it("should render the Home correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
