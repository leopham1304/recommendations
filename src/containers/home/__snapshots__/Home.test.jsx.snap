// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the Home container correctly 1`] = `
<Memo(Home)>
  <div
    className="home-container"
  >
    <Memo(NavigationBar)
      darkMode={false}
      setDarkMode={[Function]}
    >
      <nav
        className="nav-container"
      >
        <div>
          <img
            alt="AirBnB Logo"
            className="airbnb-logo"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "airbnb.svg",
              }
            }
          />
        </div>
        <div
          className="nav-menu"
        >
          <span
            className="toggle"
          >
            <input
              checked={false}
              className="checkbox"
              id="checkbox"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="checkbox"
            />
          </span>
          <span>
            Become a host
          </span>
          <span>
            Help
          </span>
          <span>
            Sign up
          </span>
          <span>
            Log in
          </span>
        </div>
      </nav>
    </Memo(NavigationBar)>
    <div
      className="autocomplete-wrapper"
    >
      <Memo(Autocomplete)
        darkMode={false}
      >
        <div
          className="autocomplete"
        >
          <input
            onChange={[Function]}
            placeholder="Try Vietnam"
            value=""
          />
        </div>
      </Memo(Autocomplete)>
    </div>
  </div>
</Memo(Home)>
`;
